src/apps/media/src/app/views/InsightsMedia.tsx

```diff
@@ -41,16 +57,17 @@ export const InsightsMedia: FC = () => {
     bins?.map((bin) => bin.id),
     { skip: !bins?.length }
   );
+
   const { data: priorUsage, isFetching: isPriorUsageFetching } =
-    useGetUsageQuery([
-      moment(date).subtract(2, "months").format(),
-      moment(date).subtract(1, "months").format(),
-    ]);
+    useGetUsageQuery([priorStartDate.format(), priorEndDate.format()]);
   const { data: usage, isFetching: isUsageFetching } = useGetUsageQuery([
-    moment(date).subtract(1, "months").format(),
-    moment(date).format(),
+    startDate.format(),
+    endDate.format(),
   ]);
 
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
```

Comment:
Is there a potential for a memory leak holding a reference to an element in state? If the button was clicked, setting state to reference the DOM element, and then the view is changed unmounting this component wouldn't this create a detached DOM node?